core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -920,8 +908,23 @@ protected SearchService newSearchService(ClusterService clusterService, IndicesS
         return customNameResolvers;
     }
 
-    /** Constructs an internal node used as a client into a cluster fronted by this tribe node. */
-    protected Node newTribeClientNode(Settings settings, Collection<Class<? extends Plugin>> classpathPlugins, Path configPath) {
+    public static class NodeBuilder {
```

Comment:
To play nicely with the `MockNode` concept used by the tests, we cannot just create the Node instance ourselves. However, by removing Guice injection, I've just replaced this class by a `BiFunction<Settings, Path>`.